geoengine/workflow.py

```diff
@@ -464,12 +465,18 @@ def save_as_dataset(
 
         return Task(TaskId.from_response(response))
 
+    # pylint: disable=too-many-locals
     async def raster_stream(
-            self,
-            query_rectangle: QueryRectangle,
-            open_timeout: int = 60) -> AsyncIterator[RasterTile2D]:
+        self,
+        query_rectangle: QueryRectangle,
+        open_timeout: int = 60,
+        bands: Optional[List[int]] = None  # TODO: move into query rectangle?
+    ) -> AsyncIterator[RasterTile2D]:
         '''Stream the workflow result as series of RasterTile2D (transformable to numpy and xarray)'''
 
+        if bands is None:
+            bands = [0]
```

Comment:
Maybe throw an error instead when the bands are empty.

Reply:
I think taking the first band is the default we want. It's only realized in this way "because of the linter", see my previous comment.

Reply:
But the user could input `band=[]` …

Reply:
886fac352fcd3fe58913a8d40fb193e617e67d7d